src/core/server/http/router/request.ts

```diff
@@ -22,10 +22,26 @@ import { ObjectType, TypeOf } from '@kbn/config-schema';
 import { Request } from 'hapi';
 
 import { filterHeaders, Headers } from './headers';
-import { RouteSchemas } from './route';
+import { RouteSchemas, RouteConfigOptions } from './route';
+
+interface KibanaRequestRoute {
+  path: string;
+  method: 'get' | 'post' | 'put' | 'patch' | 'delete' | 'options';
+  options: RouteConfigOptions;
+}
 
 /** @public */
 export class KibanaRequest<Params = unknown, Query = unknown, Body = unknown> {
+  public static getRoute(request: Request): KibanaRequestRoute {
+    return {
```

Comment:
__question:__ should this object be deeply frozen?

Reply:
`KibanaRequest` declares it as readonly

Reply:
well, `readonly` means you can't override property entirely, but you're still able to mutate the properties of this object, right?

Reply:
correct, although I'm waiting we update to the last TS version when we can use [as const](https://devblogs.microsoft.com/typescript/announcing-typescript-3-4/#const-assertions) to declare deep readonly interface.
I understand that our consumer may not use Typescript or mute errors, so I'd prefer complex solution: we should declare the whole `KibanaRequest` immutable

Reply:
I will use our own `RecursiveReadonly` for time being.